pkg/asset/agent/manifests/common.go

```diff
@@ -45,7 +46,7 @@ func getProxy(proxy *types.Proxy, machineNetwork *[]types.MachineNetworkEntry, r
 		if cidr != "" {
 			if noProxy == "" {
 				noProxy = cidr
-			} else {
+			} else if !strings.Contains(noProxy, cidr) {
```

Comment:
It would be useful to have a unit test for that.

Reply:
Added a unit test.